utils.c

```diff
@@ -343,19 +343,23 @@ float utils_fast_atan2(float y, float x) {
  * The maximum magnitude.
  *
  * @return
- * True if saturation happened, false otherwise
+ * True if saturated, false otherwise
  */
 bool utils_saturate_vector_2d(float *x, float *y, float max) {
 	bool retval = false;
-	float mag = sqrtf(SQ(*x) + SQ(*y));
-	max = fabsf(max);
 
-	if (mag < 1e-10) {
-		mag = 1e-10;
+	float mag2 = SQ(*x) + SQ(*y);
+	float max2 = SQ(max);
+
+	// Ensure magnitude doesn't become too small
+	if (mag2 < 1e-20) {
+		mag2 = 1e-20;
```

Comment:
Remove this conditional; instead document in the function documentation that `max` must be positive.